Sources/ApodiniDeploy/ApodiniDeployInterfaceExporter.swift

```diff
@@ -0,0 +1,250 @@
+//
+//  ApodiniDeployInterfaceExporter.swift
+//  
+//
+//  Created by Lukas Kollmer on 2021-01-14.
+//
+
+import Foundation
+import Apodini
+import ApodiniUtils
+import ApodiniVaporSupport
+import ApodiniDeployBuildSupport
+import ApodiniDeployRuntimeSupport
+@_implementationOnly import Vapor
+@_implementationOnly import AssociatedTypeRequirementsVisitor
+
+
+
+extension AnyEndpointParameter {
+    var lk_stableIdentity: String {
+        switch self.parameterType {
+        case .content:
+            return "c:\(self.name)"
+        case .lightweight:
+            return "l:\(self.name)"
+        case .path:
+            return "p:\(self.name)"
+        }
+    }
```

Comment:
Yes exactly. The `id`s are randomly generated when the web service is run, meaning that we cannot use a parameter's id to compare parameters across multiple instances of the same web service.
I didn't add it to `AnyEndpointParameter` because I actually wasn't sure if there is any guarantee that a `Handler` cannot have multiple `@Parameter`s with the same name.

Reply:
It can be that an Endpoint has multiple `@Parameter`s with the same name. The semantic model checks for this based on the collected `parameterNamespace` of each exporter. A normal exporter has a `.global` namespace indicating that parameter names must be unique. But some exporters might have a weaker namespace rule, e.g. the RESTful exporter. You can see the possible name spaces in `ParameterNamespace.swift`. As these name space definitions currently correspond to the main supported HTTP Options using this information and moving this into the AnyEndpointParameter should be fine 👍